examples/config-gc.json

```diff
@@ -3,10 +3,48 @@
     "storage": {
         "rootDirectory": "/tmp/zot",
         "gc": true,
-        "gcReferrers": true,
         "gcDelay": "2h",
         "untaggedImageRetentionDelay": "4h",
-        "gcInterval": "1h"
+        "gcInterval": "1h",
+        "retention": {
+            "dryRun": false,
+            "policies": [
+                {
+                    "repoNames": ["infra/*", "prod/*"],
+                    "deleteReferrers": false,
+                    "deleteUntagged": true,
```

Comment:
How does retention work? Say if repo1 has a retention policy for an image and repo2 uses an image from repo1 and has a much larger retention policy. The image from repo1 goes away but the layers are retained from a blob pov?

Reply:
what do  you mean by "uses", if by that you mean it has some blobs deduped from image1, then yes.

Reply:
These rules are for manifest and tags. This knob only deletes manifests. Manifest do no "use" other manifests (unless they are signatures or some other kind of artifacts), but that is in the context of the same repo.

Manifests could reuse layers also referenced by other manifests (base image).
Layers are GCed later, they are not "retained", and like Sebi mentioned, the dedupe functionality is cross-repo, and takes care internally of keeping a valid copy of the layer.